### PR TITLE
Add ASCII case for string versions of std.uni.toLower/toUpper

### PR DESCRIPTION
**Pull request by JackStouffer**
When profiling this function using the plain text version of Pride and Prejudice located [here](https://www.gutenberg.org/cache/epub/1342/pg1342.txt), the results are as follows:

before the changes:
```
result	52 ms, 127 μs, and 8 hnsecs
```

after

```
result	35 ms, 958 μs, and 4 hnsecs
```

### CODE REVIEW

File: std/uni.d [hunk 1]
```diff
@@ -8050,6 +8051,12 @@ private S toCase(alias indexFn, uint maxIdx, alias tableFn, S)(S s) @trusted pur
         result.reserve(s.length);
         foreach (dchar c; s[i .. $])
         {
+            if (c.isASCII)
+            {
+                result.put(asciiConvert(c));
+                continue;
+            }
+
```
Comment (wilzbach) [inline std/uni.d]: Nitpick: why not if.. else?

**Comment by wilzbach** [comment]
That's quite a lot! Great job!

Btw imho we should have such performance benchmarks with Phobos or at least run them on every merge to avoid performance regressions..

**Comment by JackStouffer** [comment]
@wilzbach fixed.

**Comment by DmitryOlshansky** [comment]
Auto-merge toggled on

**Comment by JackStouffer** [comment]
Thanks!

**Pull request merged**